blocks/blocks.go

```diff
@@ -52,14 +54,15 @@ type Config struct {
 }
 
 // NewBlockHandler creates new BlockHandler
-func NewBlockHandler(cfg Config, m mesh, v blockValidator, lg log.Log) *BlockHandler {
+func NewBlockHandler(cfg Config, m mesh, v blockValidator, lg log.Log, metrics metrics.Metrics) *BlockHandler {
```

Comment:
please no need to pass metrics to object. it would mean that we will have to follow this practice and this is just too much complexity. if you want to mock metrics for tests then  it should be done globally, but i in general doubt that this is necessary